modules/cloud-router-connection/main.tf

```diff
@@ -0,0 +1,146 @@
+terraform {
+  required_providers {
+    equinix = {
+      source  = "equinix/equinix"
+      version = ""
+    }
+  }
+}
+
+data "equinix_fabric_service_profiles" "zside" {
+    count = var.zside_ap_type == "SP" ? 1 : 0
+  filter {
+    property = "/name"
+    operator = "="
+    values   = [var.fabric_sp_name]
+  }
+}
+data "equinix_fabric_ports" "zside" {
+    count  = var.zside_ap_type == "COLO" ? 1 : 0
+  filters {
+    name   = var.zside_port_name
+  }
+}
+
+
+resource "equinix_fabric_connection" "primary_cloud_router_connection" {
+  name = var.connection_name
+  type = var.connection_type
+  notifications {
+    type   = var.notifications_type
+    emails = var.notifications_emails
+  }
+  additional_info = var.additional_info != [] ? var.additional_info : null
+  bandwidth       = var.bandwidth
+  redundancy { priority = var.redundancy }
+  order {
+    purchase_order_number = var.purchase_order_number
+  }
+  a_side {
+    access_point {
+      type = var.aside_ap_type
+      router {
+        uuid = var.fcr_uuid
+      }
+    }
+  }
+  dynamic "z_side" {
+    #ports z_side type
+    for_each = var.zside_ap_type == "COLO" ? [1] : []
+    content {
+      access_point {
+          type = var.zside_ap_type != "" ? var.zside_ap_type : null
```

Comment:
Dynamic block condition check is enough. It will override these sub checks. There's another comment on this but adding here to note this block needs that same change.